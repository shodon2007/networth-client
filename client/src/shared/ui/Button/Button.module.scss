.Button {
	font-size: var(--font-size-s-desktop);
	padding: 18px 32px;
	border-radius: var(--medium-border-radius);
	background-color: var(--bg-secondary-color);
	color: var(--primary-color);
	cursor: pointer;
}

.Button:hover {
	background-color: var(--bg-secondary-hover-color);
}
