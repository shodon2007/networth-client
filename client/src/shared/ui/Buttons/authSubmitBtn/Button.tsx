import { ButtonHTMLAttributes, FC, ReactNode, MouseEvent } from "react";
import cls from './Button.module.scss';

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
    children?: ReactNode,
    onClick: (e : MouseEvent<HTMLButtonElement>) => void,
}

const Button: FC<ButtonProps> = ({children, onClick, ...rest}) => {
    return <button className={cls.Button} onClick={onClick} {...rest}>{children}</button>
}


export default Button;