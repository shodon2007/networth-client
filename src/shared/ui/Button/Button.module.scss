.Button {
    cursor: pointer;
}

.Button.submit_button{
    border-radius: 14px;
    padding: 0px 12px;
    width: 400px;
    height: 60px;
  
    font-family: var(--font-main-Blin);
    font-weight: 600;
    font-size: var(--font-size-mm-desktop);
    color: var(--white-color);
  
    background: var(--main-black-02);
    transition: .5s;
    &:hover{
      transform: translateY(-10px) scale(1.02);
      background: var(--main-black-04);
    }
  }