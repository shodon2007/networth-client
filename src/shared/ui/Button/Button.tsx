import {ButtonHTMLAttributes, FC, ReactNode} from "react";
import classNames from "src/shared/lib/classNames";

import cls from "./Button.module.scss";

enum ThemeButton {
	CLEAR = "clear",
	SMALL = "small",
	SUBMIT_BUTTON = "submit_button",
	RED = "red",
}

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
	className?: string;
	children: ReactNode;
	theme?: ThemeButton;
}

const Button: FC<ButtonProps> = ({
	children,
	theme = "",
	className = "",
	...props
}) => {
	return (
		<button
			className={classNames(cls.Button, {}, [cls[theme], className])}
			{...props}
		>
			{children}
		</button>
	);
};

export default Button;
export {ThemeButton};
