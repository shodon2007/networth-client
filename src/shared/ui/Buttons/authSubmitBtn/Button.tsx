import {ButtonHTMLAttributes, FC, ReactNode} from "react";

import cls from "./Button.module.scss";

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
	children?: ReactNode;
}

const Button: FC<ButtonProps> = ({children, onClick, ...rest}) => {
	return (
		<button className={cls.Button} onClick={onClick} {...rest}>
			{children}
		</button>
	);
};

export default Button;
