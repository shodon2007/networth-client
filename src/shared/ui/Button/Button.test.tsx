import {render} from "@testing-library/react";
import Button, {ThemeButton} from "./Button";

describe("Testing Button.test.tsx", () => {
	test("testing default styles", () => {
		const screen = render(<Button>hello</Button>);
		const button = screen.getByText("hello");
		expect(button).toBeInTheDocument();
		expect(button).toMatchSnapshot();
	});
	describe("Testing themes", () => {
		test("testing clear theme", () => {
			const screen = render(<Button theme={ThemeButton.CLEAR}>hello</Button>);
			const button = screen.getByText("hello");
			expect(button).toBeInTheDocument();
			expect(button).toMatchSnapshot();
		});
		test("testing red theme", () => {
			const screen = render(<Button theme={ThemeButton.RED}>hello</Button>);
			const button = screen.getByText("hello");
			expect(button).toBeInTheDocument();
			expect(button).toMatchSnapshot();
		});
		test("testing small theme", () => {
			const screen = render(<Button theme={ThemeButton.SMALL}>hello</Button>);
			const button = screen.getByText("hello");
			expect(button).toBeInTheDocument();
			expect(button).toMatchSnapshot();
		});
		test("testing submit theme", () => {
			const screen = render(<Button theme={ThemeButton.SUBMIT}>hello</Button>);
			const button = screen.getByText("hello");
			expect(button).toBeInTheDocument();
			expect(button).toMatchSnapshot();
		});
	});
});
