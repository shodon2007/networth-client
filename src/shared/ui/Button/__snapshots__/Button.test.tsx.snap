// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Button.test.tsx Testing themes testing clear theme 1`] = `
<button
  class="button clear"
>
  hello
</button>
`;

exports[`Testing Button.test.tsx Testing themes testing red theme 1`] = `
<button
  class="button red"
>
  hello
</button>
`;

exports[`Testing Button.test.tsx Testing themes testing small theme 1`] = `
<button
  class="button small"
>
  hello
</button>
`;

exports[`Testing Button.test.tsx Testing themes testing submit theme 1`] = `
<button
  class="button submit"
>
  hello
</button>
`;

exports[`Testing Button.test.tsx testing default styles 1`] = `
<button
  class="button"
>
  hello
</button>
`;
